Tests/Functional/ClangModuleTests.swift

```diff
@@ -100,6 +100,29 @@ class ClangModulesTestCase: XCTestCase {
             XCTAssertFileExists(prefix, ".build", "debug", "libCDynamicLookup.so")
         }
     }
+
+    #if os(OSX)
+    func testObjectiveCModule() {
+        fixture(name: "ClangModules/Objc") { prefix in
+            XCTAssertBuilds(prefix)
+
+            // Check if objc module is present.
+            XCTAssertFileExists(prefix, ".build", "debug", "libobjc.so")
+
+            // Test objc executable (depends on objc module).
+            let objcExec = ".build/debug/objc_exec"
+            XCTAssertFileExists(prefix, objcExec)
+            var output = try popen([Path.join(prefix, objcExec)])
+            XCTAssertEqual(output, "Hello from Objc 5")
+
+            // Test swift executable (depends on objc module).
+            let swiftExec = ".build/debug/swift-exec"
+            XCTAssertFileExists(prefix, swiftExec)
+            output = try popen([Path.join(prefix, swiftExec)])
+            XCTAssertEqual(output, "5\n")
+        }
+    }
+    #endif
```

Comment:
Despite my usual emphasis on tests, I actually don't think it is worth adding the feature and everything else for this test, since it doesn't actually add fundamental new behavior. Once we have a mechanism for cheap "convention" tests we can just check we recognize the file, but there is no need to built Obj-C specifically at least as part of this functional change.